File: Problem Solving/Algorithms/Strings/Pangrams.py
#!/bin/python3

import math
import os
import random
import re
import sys
from string import ascii_lowercase
from string import ascii_uppercase

#
# Complete the 'pangrams' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def pangrams(s):
    # Write your code here
    letter_map = {" ": 26}
    for i in range(26):
        letter_map[ascii_lowercase[i]] = i
        letter_map[ascii_uppercase[i]] = i

    seen = [0] * 27
    for char in s:
        seen[letter_map[char]] = 1

    del seen[-1]
    if all(seen):
        return "pangram"
    return "not pangram"



if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = pangrams(s)

    fptr.write(result + '\n')

    fptr.close()

File: Problem Solving/Algorithms/Strings/Pangrams.cpp
#include <bits/stdc++.h>

using namespace std;

/*
 * Complete the 'pangrams' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

string pangrams(string s) {
    vector<bool> seen(25, false);
    for (auto ch: s) {
        int ch_idx;
        if (int(ch) >= 65 && int(ch) <= 90)
            ch_idx = int(ch) - 65;
        else if (int(ch) >= 97 && int(ch) <= 122)
            ch_idx = int(ch) - 97;
        else
            continue;

        seen[ch_idx] = true;
    }

    bool is_all_true = all_of(
        seen.begin(), seen.end(),
        [](bool elem){return elem == true;});
    if (is_all_true)
        return "pangram";
    return "not pangram";
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    string result = pangrams(s);

    fout << result << "\n";

    fout.close();

    return 0;
}
